README.md

```diff
@@ -6,3 +6,52 @@
 [![Coverage](https://codecov.io/gh/invenia/FeatureDescriptors.jl/branch/main/graph/badge.svg)](https://codecov.io/gh/invenia/FeatureDescriptors.jl)
 [![Code Style: Blue](https://img.shields.io/badge/code%20style-blue-4495d1.svg)](https://github.com/invenia/BlueStyle)
 [![ColPrac: Contributor's Guide on Collaborative Practices for Community Packages](https://img.shields.io/badge/ColPrac-Contributor's%20Guide-blueviolet)](https://github.com/SciML/ColPrac)
+
+FeatureDescriptors.jl is an interface package for describing features used in [models](https://invenia.github.io/Models.jl/stable/), [feature engineering](https://invenia.github.io/FeatureTransforms.jl/stable/), and other data-science workflows.
+
+## Getting Started
+
+A `Descriptor` is an abstract type that provides a way to describe the properties and relationships for a collection of features at a high-level.
+`Descriptor` supertypes define some general properties, such as the required tables or value column, that can be overloaded for specific
+subtypes as required. 
+
+For example, say that some weather data is contained in `weather.csv` file but each column describes a different feature we are interested in, such as `:temperature` or `:humidity`. 
+We can define a general `Weather` `Descriptor` with subtypes `Temperature` and `Humidity` that are loaded from the same table but using the appropriate columns:
+
+```julia
+using FeatureDescriptors
+
+abstract type Weather <: Descriptor end
+
+FeatureDescriptors.tables(::Type{<:Weather}) = ["weather.csv"]
+FeatureDescriptors.value_col(::Type{<:Weather}) = :weather  # dummy
+
+abstract type Temperature <: Weather end
+FeatureDescriptors.value_col(::Type{Temperature}) = :temperature
+
+abstract type Humidity <: Weather end
+FeatureDescriptors.value_col(::Type{Humidity}) = :humidity
+
+```
+
+A more specific instance of a feature can also be defined, such as a `MeanTemperature`, perhaps if that feature requires a different [feature engineering](https://invenia.github.io/FeatureTransforms.jl/stable/) treatment.
+```julia
+abstract type MeanTemperature <: Temperature end
+```
+
+Finally, other useful features might be contained in a different table entirely, but we may stil want to encode its relationship to the others.
+
+```julia
+abstract type Precipitation <: Weather end
+FeatureDescriptors.tables(::Type{<:Precipitation}) = ["rainfall.csv"]
+FeatureDescriptors.value_col(::Type{<:Precipitation}) = :rainfall 
+```
+
+## The Descriptors API
+
+All `Descriptor`s are required to implement the following:
+
+1. A `tables` method that specifies the table(s) containing the data. Note: a `Descriptor` may be associated with multiple tables, because it may be necessary to perform [feature engineering](https://invenia.github.io/FeatureTransforms.jl/stable/) to create the derived feature.
```

Comment:
might we want to change this to something more generic like `source`?

Reply:
I think so. It looks high-level enough to not have an opinion about the data structure.

Should `value_col` change along with? `value_key`, `value_entry`?

Reply:
other alternatives: `value_name`, `value_label`, `value_handle`

Reply:
or simply just `sources`, `value` and `categories`? (although `value` is probably too close to `values`)

Reply:
Or `quantity` and `categories`. That would then be consistent with the language around describing [variable types](http://sites.utexas.edu/sos/variables/)

Reply:
`quantity` sounds like you'd be getting a single number back. My preference is `value_key`.

Reply:
how about `quantitative_key` and `categorical_keys`?

Reply:
> how about `quantity_key` and `category_keys`?

I prefer to be less technical and not to have so many syllables.

Why not just `key`? It's generic but I think it suits. There's a `keys` method for Dicts and DataFrames and general iterators of key-value pairs, so it would be associated with that.

Otherwise, I still think `value_key` is good but it's a bit confusing how it sounds like the reverse of "key-value". `value_name` sounds good to me too.

By the way, which thing is `category_keys` or `categories` replacing?

Reply:
>I prefer to be less technical and not to have so many syllables.
>Why not just key? 

The thing about having `key` (especially if we associate with `Base.keys`) is it might imply there's a corresponding `values` method. But we couldn't implement that because it would likely result in type piracy.

But if we ignore the `values` technicality, I still feel `key` is almost too generic having thought about it. It wouldn't seem obvious (to me at least) what part it plays in the context of features. Whereas qualifying it with `quantity` at least connects it to terms used for statistical data types. 

But maybe that seems too technical to you?

>Otherwise, I still think value_key is good but it's a bit confusing how it sounds like the reverse of "key-value". value_name sounds good to me too.

this is a good point as well

>By the way, which thing is category_keys or categories replacing?

Currently it's an internal method. I was going to add it later but it should probably go in this PR.

Reply:
I'm ok to rule out `key`, those are fair points.

> But maybe that seems too technical to you?

Yes. `quantity` might be the most accurate in one technical sense but I'd confuse it with a single number and it's more of a mouthful.

Reply:
what about `quantity_key`? that's what I meant by "qualify it with..." but perhaps that wasn't clear.

Reply:
Sorry I was thinking of `quantity_key` as well when I said that.

To be clear, I might not even think twice if it was `quantity_key` and wasn't opened for discussion in the first place. Take these preferences into account, but I wouldn't let it block you.

Reply:
fair enough, I went with `quantity_key` in the end. Hopefully it's not too controversial.